x/vpool/types/pool.go

```diff
@@ -67,56 +75,43 @@ args:
   - baseAmount: the amount of base asset to add to/remove from the pool
 
 ret:
-  - quoteAmountOut: the amount of quote assets required to make this hypothetical swap
+  - quoteOutAbs: the amount of quote assets required to make this hypothetical swap
     always an absolute value
   - err: error
 */
 func (vpool *Vpool) GetQuoteAmountByBaseAmount(
-	dir Direction, baseAmount sdk.Dec,
-) (quoteAmount sdk.Dec, err error) {
-	if baseAmount.IsZero() {
+	baseDelta sdk.Dec,
+) (quoteOutAbs sdk.Dec, err error) {
+	if baseDelta.IsZero() {
 		return sdk.ZeroDec(), nil
 	}
 
 	invariant := vpool.QuoteAssetReserve.Mul(vpool.BaseAssetReserve) // x * y = k
 
-	var baseAssetsAfter sdk.Dec
-	if dir == Direction_ADD_TO_POOL {
-		baseAssetsAfter = vpool.BaseAssetReserve.Add(baseAmount)
-	} else {
-		baseAssetsAfter = vpool.BaseAssetReserve.Sub(baseAmount)
-	}
-
-	if baseAssetsAfter.LTE(sdk.ZeroDec()) {
+	baseReservesAfter := vpool.BaseAssetReserve.Add(baseDelta)
+	if baseReservesAfter.LTE(sdk.ZeroDec()) {
 		return sdk.Dec{}, ErrBaseReserveAtZero.Wrapf(
 			"base assets below zero after trying to swap %s base assets",
-			baseAmount.String(),
+			baseDelta.String(),
 		)
 	}
 
-	quoteAssetsAfter := invariant.Quo(baseAssetsAfter)
-	quoteAmount = quoteAssetsAfter.Sub(vpool.QuoteAssetReserve).Abs()
-
-	return quoteAmount, nil
-}
+	quoteReservesAfter := invariant.Quo(baseReservesAfter)
+	quoteOutAbs = quoteReservesAfter.Sub(vpool.QuoteAssetReserve).Abs()
 
-// IncreaseBaseAssetReserve increases the quote reserve by amount
-func (vpool *Vpool) IncreaseBaseAssetReserve(amount sdk.Dec) {
-	vpool.BaseAssetReserve = vpool.BaseAssetReserve.Add(amount)
+	return quoteOutAbs, nil
 }
 
-// DecreaseBaseAssetReserve descreases the quote asset reserve by amount
-func (vpool *Vpool) DecreaseBaseAssetReserve(amount sdk.Dec) {
-	vpool.BaseAssetReserve = vpool.BaseAssetReserve.Sub(amount)
-}
-
-func (vpool *Vpool) IncreaseQuoteAssetReserve(amount sdk.Dec) {
```

Comment:
Using both an Increase and Decrease function didn't make sense to me since we can just pass in a negative number and have code that's easier to understand in the swap function(s)